zerver/tests/test_project.py

```diff
@@ -0,0 +1,28 @@
+import importlib
+import os
+from unittest import mock
+
+from zerver.lib.test_classes import ZulipTestCase
+from zproject import config
+
+
+class ConfigTest(ZulipTestCase):
+    def test_get_mandatory_secret_succeed(self) -> None:
+        secret = config.get_mandatory_secret("shared_secret")
+        self.assertGreater(len(secret), 0)
+
+    def test_get_mandatory_secret_failed(self) -> None:
+        with self.assertRaisesRegex(config.ZulipSettingsError, "nonexistent"):
+            config.get_mandatory_secret("nonexistent")
+
+    def test_disable_mandatory_secret_check(self) -> None:
+        # Because we mock the environment variable, which affects how zproject.config.DISABLE_MANDATORY_SECRET_CHECK
+        # is set, we need to reload the module to see the change.
+        try:
+            with mock.patch.dict(os.environ, {"DISABLE_MANDATORY_SECRET_CHECK": "True"}):
+                importlib.reload(config)
+                secret = config.get_mandatory_secret("nonexistent")
+        finally:
+            # To avoid polluting the rest of the test case, reload the module again without the mock.
+            importlib.reload(config)
+        self.assertEqual(secret, "")
```

Comment:
Would it be better to just check the environment variable directly in `get_mandatory_secret`, rather than caching the value in a variable and needing to do messy test? 

Another option would be to use `with self.settings(DISABLE_MANDATORY_SECRET_CHECK=True)` for this test, rather than the `importlib` stuff.

I think we can also call this test file `test_server_settings.py`, for contrast with `test_settings.py`, which we may want to rename to `test_user_settings.py` at some point.

Reply:
Because we can't make `DISABLE_MANDATORY_SECRET_CHECK` a setting due to cyclic dependencies, having an inline check for the environment variable would be better.